lib/data_stores/mysql_advanced_data_store.py

```diff
@@ -26,23 +26,48 @@ class Error(data_store.Error):
   """Base class for all exceptions in this module."""
 # pylint: enable=nonstandard-exception
 
+class SafeQueue(Queue.Queue):
+  def __init__(self, maxsize=0):
```

Comment:
I don't think that will work be cause immediately after self.mutext=threading.RLock() it starts using self.mutex for other initialization stuff.  I think the right way to fix this is for Queue itself to be using RLock, but patching python to make GRR work isn't much of an option.